### PR TITLE
Update kfp sdk dependency to 1.8.10+

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
- Python 3.6 is no longer supported officially. 
- Update kfp sdk dependency to 1.8.10+
- New KFP DSL has new python string formatting syntax, but no functional changes
- Update KFP SDK dependency chain and unit tests

There are some pointers to kfp-tekton 1.2.0. However, there are still some features we are pending so we are not yet cutting the new SDK release.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/tests/README.md [hunk 1]
```diff
@@ -27,7 +27,12 @@ You should see an output similar to the one below, outlining which test scripts
 which are failing:
 
 ```YAML
-KFP SDK version: 1.7.2
+KFP clone version: 1.8.0
+KFP Python SDK version(s):
+kfp                      1.8.11
+kfp-pipeline-spec        0.1.13
+kfp-server-api           1.8.1
+kfp-tekton               1.1.1     /Users/tommyli/Desktop/kfp-sdk/sdk/python
```
Comment (ScrapCodes) [inline sdk/python/tests/README.md:35]: `/Users/tommyli/Desktop/kfp-sdk/sdk/python` an oversight? or by design?

**Comment by Tomcli** [comment]
E2E CI for this PR: https://cloud.ibm.com/devops/pipelines/tekton/f85a45d2-36fc-45fb-a449-b01c13cf5191/runs/7b32b6ff-c940-4809-ac6b-5cacdb040102/test/run-go-unittests?env_id=ibm:yp:us-south

**Comment by Tomcli** [comment]
@akchinSTC are you okay with only supporting kfp 1.8.10+?

**Review by yhwang — APPROVED** [review]

**Pull request merged**